httpserver/middleware/metrics_params.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package middleware
+
+// MetricsParams stores parameters for the HTTPRequestMetrics middleware
+// that may be modified dynamically by the other underlying middlewares/handlers.
+type MetricsParams struct {
+	labels map[string]string
+}
+
+// AddLabel adds a new label with the specified name and value.
+// If a label with the same name already exists, it will be overwritten.
+func (mp *MetricsParams) AddLabel(name, value string) {
+	if mp.labels == nil {
+		mp.labels = make(map[string]string)
+	}
+	mp.labels[name] = value
+}
+
+// GetLabels returns a copy of all labels.
+func (mp *MetricsParams) GetLabels() map[string]string {
+	if mp.labels == nil {
+		return nil
+	}
+	labels := make(map[string]string, len(mp.labels))
```

Comment:
Why do we need to create a deep copy here?
Actually, I'm not sure we need this method at all. Will it be used in the client code? In what cases?